.github/workflows/integration_tests.yaml

```diff
@@ -10,14 +10,41 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+
+      # To speed up the tests, we build the charm with Charmcraft and put the
+      # `.charm` file in the repository root directory. Integration tests will
+      # pick up this file and avoid building it per test case.
+      - name: Install Charmcraft
+        run: sudo snap install charmcraft --classic
+      - name: Build charm
+        run: charmcraft pack --destructive-mode --verbosity=trace
+
       - name: Setup operator environment
         uses: charmed-kubernetes/actions-operator@main
         with:
-          juju-channel: 3.3/stable
+          channel: 1.28-strict/stable
+          juju-channel: 3.4/stable
           provider: microk8s
+          microk8s-group: snap_microk8s
           microk8s-addons: "ingress storage dns rbac registry"
-          channel: 1.25-strict/stable
+      # Instructions from https://microk8s.io/docs/registry-private
+      # and https://github.com/containerd/cri/blob/master/docs/registry.md
+      # Bash explanation https://stackoverflow.com/questions/15429330/how-to-specify-a-multi-line-shell-variable#comment21828145_15429426
+      - name: Setup microk8s for private registry
+        run: |
+          read -r -d '' REGISTRY_CONFIG << EOL || true
```

Comment:
Can we make this a composite action so we can share it across repos, seems very useful

Reply:
Sure we can. @kian99 has already done this for JIMM k8s charm. If we extract it as a separate action, then we can use it both in JIMM and Livepatch. Anyway, this is out of the scope of this PR, I think.

Reply:
I think a better approach would be to make a contribution to the https://github.com/canonical/charming-actions repo and provide the upload-charm action with extra parameters to be able to pull from private registries. Very do-able and I started down that road but it will take more time.